src/pages/SearchPage/index.js

```diff
@@ -0,0 +1,181 @@
+        return {...options, headerMessage: header};
+    }, [debouncedSearchValue, reports, personalDetails, betas, isScreenTransitionEnd]);
+
+    const sections = useMemo(() => {
+        const newSections = [];
+        let indexOffset = 0;
+
+        if (recentReports.length > 0) {
+            newSections.push({
+                data: recentReports,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (localPersonalDetails.length > 0) {
+            newSections.push({
+                data: localPersonalDetails,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (userToInvite) {
+            newSections.push({
+                data: [userToInvite],
+                shouldShow: true,
+                indexOffset,
+            });
+        }
+
+        return newSections;
+    }, [localPersonalDetails, recentReports, userToInvite]);
+
+    const selectReport = (option) => {
+        if (!option) {
+            return;
+        }
+
+        if (option.reportID) {
+            setSearchValue('');
+            Navigation.dismissModal(option.reportID);
+        } else {
+            Report.navigateToAndOpenReport([option.login]);
+        }
+    };
+
+    const handleScreenTransitionEnd = () => {
+        setIsScreenTransitionEnd(true);
+    };
+
+    const isOptionsDataReady = useMemo(() => ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(personalDetails), [personalDetails]);
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={SearchPage.displayName}
+            onEntryTransitionEnd={handleScreenTransitionEnd}
+        >
+            {({didScreenTransitionEnd, safeAreaPaddingBottomStyle}) => (
+                <>
+                    <HeaderWithBackButton title={translate('common.search')} />
+                    <View style={[themeStyles.flex1, themeStyles.w100, safeAreaPaddingBottomStyle]}>
+                        <SelectionList
+                            sections={didScreenTransitionEnd && isOptionsDataReady ? sections : []}
+                            textInputValue={searchValue}
+                            textInputLabel={translate('optionsSelector.nameEmailOrPhoneNumber')}
+                            textInputHint={offlineMessage}
+                            onChangeText={onChangeText}
+                            headerMessage={headerMessage}
+                            onLayout={searchRendered}
+                            autoFocus
+                            onSelectRow={selectReport}
+                            showLoadingPlaceholder={!didScreenTransitionEnd || !isOptionsDataReady}
+                            footerContent={<SearchPageFooter />}
```

Comment:
You can follow on https://github.com/Expensify/App/pull/31479#discussion_r1424209799 - I'd reuse this component's instance instead of passing in a new React element